tests/main/system-usernames/task.yaml

```diff
@@ -0,0 +1,760 @@
+summary: ensure seccomp rules for snap_daemon user work correctly
+
+environment:
+    # This is used to abbreviate some of the paths below.
+    P1: /var/snap/test-snapd-sh/common/test.chown
+    P2: /var/snap/test-snapd-daemon-user/common/test.chown
+    P3: /var/snap/test-snapd-daemon-user/common/test.chown32
+    # List of expected snap install failures due to libseccomp/golang-seccomp
+    # being too old. This should only reduce with time since new systems should
+    # have newer libseccomp and golang-seccomp
+    EXFAIL: "amazon-linux-2-64 centos-7-64 debian-9-64 debian-sid-64 fedora-29-64 fedora-30-64 opensuse-15\\.0-64 opensuse-15\\.1-64 ubuntu-14"
+
+prepare: |
+    echo "Install helper snaps with default confinement"
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+restore: |
+    # make sure this snap is removed in case it was installed
+    snap remove test-snapd-daemon-user || true
+
+    # snapd will create this for us, but we'll remove it for consistency in
+    # test runs
+    userdel --force snap_daemon || true
```

Comment:
do we need to specify --extra-users on core?

Reply:
I think so.